Symfony/CS/Tests/Fixer/Symfony/TrimArraySpacesFixerTest.php

```diff
@@ -0,0 +1,212 @@
+                "<?php \$foo = [ \$y ? true : false ];",
+            ),
+
+            array(
+                "<?php \$foo = array(array('foo'), array('bar'));",
+                "<?php \$foo = array( array( 'foo' ), array( 'bar' ) );",
+            ),
+
+            array(
+                "<?php \$foo = [['foo'], ['bar']];",
+                "<?php \$foo = [ [ 'foo' ], [ 'bar' ] ];",
+            ),
+
+            array(
+                "<?php function(array \$foo = array('bar')) {}",
+                "<?php function(array \$foo = array( 'bar' )) {}",
+            ),
+
+            array(
+                "<?php function(array \$foo = ['bar']) {}",
+                "<?php function(array \$foo = [ 'bar' ]) {}",
+            ),
+
+            array(
+                "<?php \$foo = array(function() {return 'foo';});",
+                "<?php \$foo = array( function() {return 'foo';} );",
+            ),
+
+            array(
+                "<?php \$foo = [function() {return 'foo';}];",
+                "<?php \$foo = [ function() {return 'foo';} ];",
+            ),
+
+            array(
+                "<?php \$foo = array(\$bar->method(), Foo::doSomething());",
+                "<?php \$foo = array( \$bar->method(), Foo::doSomething() );",
+            ),
+
+            array(
+                "<?php \$foo = [\$bar->method(), Foo::doSomething()];",
+                "<?php \$foo = [ \$bar->method(), Foo::doSomething() ];",
+            ),
+
+            array(
+                "<?php \$foo = array(array('foo'), array('bar'));",
+                "<?php \$foo = array( array('foo'), array('bar') );",
+            ),
+
+            array(
+                "<?php \$foo = [['foo'], ['bar']];",
+                "<?php \$foo = [ ['foo'], ['bar'] ];",
+            ),
+
+            array(
+                "<?php \$foo = array(array('foo'), array('bar'));",
+                "<?php \$foo = array(array( 'foo' ), array( 'bar' ));",
+            ),
+
+            array(
+                "<?php \$foo = [['foo'], ['bar']];",
+                "<?php \$foo = [[ 'foo' ], [ 'bar' ]];",
+            ),
+
+            array(
+                "<?php \$foo = array(array('foo'), array('bar'));",
+                "<?php \$foo = array( array( 'foo' ), array( 'bar' ) );",
+            ),
+
+            array(
+                "<?php \$foo = [['foo'], ['bar']];",
+                "<?php \$foo = [ [ 'foo' ], [ 'bar' ] ];",
+            ),
+
+            array(
+                "<?php \$foo = array(array('foo'), array('bar'));",
+            ),
+
+            array(
+                "<?php \$foo = [['foo'], ['bar']];",
+            ),
+
+            array(
+                "<?php \$foo = array(/* empty array */);",
+                "<?php \$foo = array( /* empty array */ );",
+            ),
+
+            array(
+                "<?php \$foo = [/* empty array */];",
+                "<?php \$foo = [ /* empty array */ ];",
```

Comment:
We need a more complex mix of a load of different stuff. Nested arrays, with different syntax, and as much crazy stuff as possible. This very separate test cases don't quite cut it. We need some more "real life" cases.